sockeye/training.py

```diff
@@ -393,8 +393,12 @@ def _fit(self,
                     checkpoint_state = CheckpointState(checkpoint=train_state.checkpoint, metric_val=m_val)
                     optimizer.pre_update_checkpoint(checkpoint_state)
 
+                # learning rate adjustment
                 if self.lr_scheduler is not None:
-                    self.lr_scheduler.new_evaluation_result(has_improved)
+                    lr_adjusted = self.lr_scheduler.new_evaluation_result(has_improved)
+                    if lr_adjusted and not has_improved:
```

Comment:
could we make this behavior optional? e.g. in a async sgd setup one might not want to reload the params here. We could make it the new default but have an option to turn it off.

Reply:
done